### PR TITLE
Added Redis as a state store backend

### PR DESCRIPTION
**Pull request by adilkhash**
## Description
Added Redis backend to store Luigi's tasks state

## Motivation and Context
We run our infrastructure inside containers hence are required to have centralized state store location

## Have you tested this? If so, how?
I ran my jobs with this code and it works for me. If community finds this PR useful I will add test.

### CODE REVIEW

**Comment by dlstadther** [comment]
This seems like a reasonable contribution. I'm not sure what the adoption rate would be. But i'm not immediately opposed to such a configurable Redis state store.

**Comment by dhuckins** [comment]
+1 would like to have this feature

**Comment by dhuckins** [comment]
am trying this in a distributed environment in front of a load balancer am getting:
`This progress looks :| because there were tasks that were not granted run permission by the scheduler`
is there a way to control how often the state is persisted?

**Comment by dhuckins** [comment]
so using redis as the backend it would need to flush every server request, that's a lot of overhead

File: luigi/scheduler.py [hunk 1]
```diff
@@ -668,6 +673,50 @@ def disable_workers(self, worker_ids):
             worker.tasks.clear()
 
 
+class SimpleTaskRedisState(SimpleTaskState):
```
Comment (Tarrasch) [inline luigi/scheduler.py]: Is it still Simple? Maybe RedisTaskState?

**Comment by adilkhash** [comment]
Closing PR because there is no reason to store dumped state, because it could be obsolete or corrupted (due to pickle)

**Pull request closed without merging**